pkg/apis/hub/v1alpha1/ai_service.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright (C) 2022-2024 Traefik Labs
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published
+by the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AIService  defines AI Service.
+type AIService struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired behavior of this APIPlan.
+	Spec AIServiceSpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+
+// AIServiceSpec the plugin configuration.
+type AIServiceSpec struct {
+	Anthropic   *Anthropic   `json:"anthropic,omitempty"`
+	AzureOpenAI *AzureOpenAi `json:"azureOpenAi,omitempty"`
+	Bedrock     *Bedrock     `json:"bedrock,omitempty"`
+	Cohere      *Cohere      `json:"cohere,omitempty"`
+	Gemini      *Gemini      `json:"gemini,omitempty"`
+	Mistral     *Mistral     `json:"mistral,omitempty"`
+	Ollama      *Ollama      `json:"ollama,omitempty"`
+	OpenAI      *OpenAi      `json:"openAi,omitempty"`
```

Comment:
OpenAI is usually referred as `openai` in the competition:
- `https://api.unify.ai/v0/models?provider=openai'`
- https://docs.llamaindex.ai/en/stable/examples/llm/openai/
- https://developers.cloudflare.com/ai-gateway/providers/openai/
- https://docs.konghq.com/hub/kong-inc/ai-proxy/how-to/llm-provider-integration-guides/openai/

I think it would be best to stick to that naming.